shopinvader_product_sale_packaging/schemas/product_packaging.py

```diff
@@ -13,6 +15,12 @@ class SimpleProductPackaging(StrictExtendableBaseModel):
     is_unit: bool
     can_be_sold: bool
 
+    def __init__(self, /, **data: Any) -> None:  # type: ignore
+        # Ensure `barcode` is set to None when empty, no matter if we use `from_packaging`.
+        if "barcode" in data and not data["barcode"]:
+            data["barcode"] = None
+        super().__init__(**data)
```

Comment:
It's not the right way to do it with pydantic. You should use the [field_validator](https://docs.pydantic.dev/latest/concepts/validators/#field-validators) method.
```suggestion
from pydantic import field_validator


        @field_validator("barcode",  mode="before")
        @classmethod
        def barcode_none_when_empty(cls, v: Any):
            # Ensure `barcode` is set to None when empty, no matter if we use `from_packaging`.
             return v or None
```